dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/PublishingEndPoint.java

```diff
@@ -145,4 +153,36 @@ public StringBuilder toURL() {
      * translated in upper layer.
      */
     public abstract void validatePublishingEndPoint() throws PublishingEndPointValidationException;
+
+    public boolean hasAuthKey() {
+        final StringBuilder authKey = getAuthKey();
+
+        if (UtilMethods.isEmpty(authKey.toString())) {
+            return false;
+        }
+
+        return !isTokenExpired() && !isTokenInvalid();
+    }
+
+    public boolean isTokenExpired() {
+        final StringBuilder authKey = getAuthKey();
+
+        if (UtilMethods.isEmpty(authKey.toString())) {
```

Comment:
any toString is creating a new String everything it is being used, it would be better to save it into a value to avoid new creations